nixos/modules/services/networking/firewall.nix

```diff
@@ -493,6 +508,22 @@ in
         '';
     };
 
+    networking.firewall.extraAllowed = mkOption {
+      type = types.listOf (types.submodule { options = {
+           source = mkOption { type = types.string; default = ""; example = "example.com"; description = "Source of the packets"; };
+           protocol = mkOption { type = types.enum ["tcp" "udp" "icmp"]; default = ""; example = "tcp"; description = "Protocol used."; };
+           dport = mkOption { type = types.int; default = 0; example = 22; description = "Destination port of the packet."; };
+           useIPv4 = mkOption { type = types.bool; default = true; description = "Use the IPv4 address of this host."; };
+           useIPv6 = mkOption { type = types.bool; default = true; description = "Use the IPv6 address of this host."; };
+      };});
+      default = [];
+      example = [ { source = "example.com"; protocol = "tcp"; dport = 22; } ];
```

Comment:
I had not realized that, but this felt wrong indeed.

The reason for the `useIPv4` and `useIPv6` is that the iptables rules introduced by this pull request might match addresses specified in either IPv4 or IPv6.

Since some hosts don't have an IPv6 yet, or don't have an IPv4 anymore, we need a way to tell whether to call iptables or ip6tables.